runtime/bcutil/cfreader.c

```diff
@@ -1074,10 +1075,10 @@ readPool(J9CfrClassFile* classfile, U_8* data, U_8* dataEnd, U_8* segment, U_8*
 			index += size;
 			if (0 == classfile->firstUTF8CPIndex) {
 				classfile->firstUTF8CPIndex = i;
-				previousUTF8 = info;
+				*previousUTF8 = info;
 			} else {
-				previousUTF8->nextCPIndex = (U_16)i;
-				previousUTF8 = info;
+				(*previousUTF8)->nextCPIndex = (U_16)i;
+				*previousUTF8 = info;
```

Comment:
Can you modify this to only write to the out pointer at the end of the function?  Writing to a local is usually cheaper than derefing a pointer in a loop.

See the code for writing `*pFreePointer = freePointer;`